### DISCUSSION TITLE
다른 스크립트를 참조하는 것이 가능한가요?

### DISCUSSION THREAD

**Discussion by JaeHyeog** [naverz/zepeto-studio-kor#893] [5. Q&A]
월드 내에서 오브젝트를 클릭하면
해당 오브젝트의 이름이 UI에서 뜨게 끔 개발을 하려고 합니다.

유니티에서는 
GameObject.Find("MainCanvas").GetComponent<mainScript>();
이런식으로 해서 스크립트를 참조해서 해당 스크립트에 있는 함수를 사용할 수 있었는데

타입스크립트에서도 사용이 가능한지 궁금합니다.
같은 방법으로는 안되는 것 같아서요

검색을 해봐도 잘 모르겠네요 ㅠㅠ 좋은 방법이 있으면 공유 부탁드리겠습니다.

**Comment by minjujuu** [comment]
타입스크립트에서도 GetComponent해서 참조할 수 있습니다.
```
Start()
   {
       // add script component
       this.gameObject.AddComponent<ExtraComponent>();
       this.extComponent = this.gameObject.GetComponent<ExtraComponent>();
   }
```
https://studio.zepeto.me/kr/guides/script-import
자세한 사항은 위 가이드를 참고하시면 도움이 될 것 같습니다!

**Reply by JaeHyeog** [reply]
오 테스트해보겠습니다! 감사합니다.

**Reply by JaeHyeog** [reply]
import { ZepetoScriptBehaviour } from 'ZEPETO.Script'
import { Input, GameObject } from "UnityEngine";
import MainCanvasScript from './MainCanvasScript';

export default class EventTriggerSample extends ZepetoScriptBehaviour {

    private m_root_script: MainCanvasScript; 

    Start() {
        this.m_root_script = GameObject.Find("MainCanvas").GetComponent<MainCanvasScript>();
    }

    Update() {
        if (Input.GetMouseButtonDown(0)) {
            this.m_root_script.SetClickName(this.name);
        }
    }
}


이런식으로 구현하면 되네요
답변 감사드립니다.
